hack/generate/samples/internal/helm/memcached.go

```diff
@@ -62,6 +61,8 @@ func (mh *MemcachedHelm) Run() {
 		os.Exit(1)
 	}
 
+	os.Setenv("KUBECONFIG", "broken_so_we_generate_static_default_rules")
```

Comment:
```suggestion
        // For Helm, when there is a cluster available, the role generator will template the default chart, lookup each of 
        // the resources from the generated manifest using the discovery API, and then generate a rule set that is 
        // tailored to those resources. The following ENV VAR will allow the sample to be generated idempotently 
        // which means with all default roles.
	os.Setenv("KUBECONFIG", "broken_so_we_generate_static_default_rules")
```

Note that if we remove it then will be harder for a developer to contribute to the project since it has diff results when the kind is or not running locally. We might be able to solve it by adding a validation or at least a WARN in the samples generate for Helm such as:

"Ensure that you have no cluster running". Also, I think we can keep a comment with more info #3873 in this part in the code to clarifies why it is required.